src/ucp/wireup/select.c

```diff
@@ -877,20 +888,19 @@ static ucs_status_t ucp_wireup_add_am_bw_lanes(ucp_ep_h ep, const ucp_ep_params_
         criteria.local_iface_flags |= UCP_WORKER_UCT_UNSIG_EVENT_CAP_FLAGS;
     }
 
-    status        = UCS_ERR_UNREACHABLE;
-    num_lanes     = 1;
-    tl_bitmap     = -1;
-    remote_md_map = -1;
-    md_map        = 0;
+    status             = UCS_ERR_UNREACHABLE;
+    num_lanes          = 1;
+    local_dev_bitmap   = -1;
+    remote_dev_bitmap  = -1;
+    md_map             = 0;
 
     /* am_bw_lane[0] is am_lane, so don't re-select it here */
     for (lane_desc_idx = 0; lane_desc_idx < *num_lanes_p; ++lane_desc_idx) {
         if (lane_descs[lane_desc_idx].usage & UCP_WIREUP_LANE_USAGE_AM) {
-            addr_index     = lane_descs[lane_desc_idx].addr_index;
-            rsc_index      = lane_descs[lane_desc_idx].rsc_index;
-            remote_md_map &= ~UCS_BIT(address_list[addr_index].md_index);
-            md_map        |= UCS_BIT(context->tl_rscs[rsc_index].md_index);
-            tl_bitmap      = ucp_wireup_unset_tl_by_md(ep, tl_bitmap, rsc_index);
+            addr_index        = lane_descs[lane_desc_idx].addr_index;
+            rsc_index         = lane_descs[lane_desc_idx].rsc_index;
+            md_map           |= UCS_BIT(context->tl_rscs[rsc_index].md_index);
```

Comment:
i guess we don't need to deselect by md_map now?
can you pls test that it works on a machine with dual-port HCA such as ConnectX-3 or Connect-IB?

Reply:
md_map here is used to limit number of lanes by MD's
tested on Connect-X3 (single HCA with double port) - number of lanes has no effect to results (single lane is always selected)

Reply:
correction:
ucx_info say:
  lane[0]: 3:rc/mlx4_0:1 md[1]           -> md[1] rma_bw#0 am am_bw#0
  lane[1]: 4:rc/mlx4_0:2 md[1]           -> md[1] rma_bw#1 am_bw#1 wireup{ud/mlx4_0:1}

it seems different ports have different device numbers

Reply:
connect-X3 is PCIe 8x so should not have large difference with FDR
in order to test can degrade port speed to QDR